File: src/extras-knewstuff/libcategorymetadata.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KNSCore__CategoryMetadata
#include <QString>
#include <QByteArray>
#include <cstring>
#include <categorymetadata.h>
#include "libcategorymetadata.h"
#include "libcategorymetadata.hxx"

KNSCore__CategoryMetadata* KNSCore__CategoryMetadata_new(const KNSCore__CategoryMetadata* param1) {
    return new KNSCore::CategoryMetadata(*param1);
}

libqt_string KNSCore__CategoryMetadata_Id(const KNSCore__CategoryMetadata* self) {
    QString _ret = self->id();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KNSCore__CategoryMetadata_Name(const KNSCore__CategoryMetadata* self) {
    QString _ret = self->name();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KNSCore__CategoryMetadata_DisplayName(const KNSCore__CategoryMetadata* self) {
    QString _ret = self->displayName();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KNSCore__CategoryMetadata_Delete(KNSCore__CategoryMetadata* self) {
    delete self;
}

File: src/extras-knewstuff/libcategorymetadata.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/knscore-categorymetadata.html
pub const knscore__categorymetadata = struct {
    /// New constructs a new KNSCore::CategoryMetadata object.
    ///
    /// ``` param1: QtC.KNSCore__CategoryMetadata ```
    pub fn New(param1: ?*anyopaque) QtC.KNSCore__CategoryMetadata {
        return qtc.KNSCore__CategoryMetadata_new(@ptrCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/knscore-categorymetadata.html#id)
    ///
    /// ``` self: QtC.KNSCore__CategoryMetadata, allocator: std.mem.Allocator ```
    pub fn Id(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KNSCore__CategoryMetadata_Id(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("knscore::categorymetadata.Id: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/knscore-categorymetadata.html#name)
    ///
    /// ``` self: QtC.KNSCore__CategoryMetadata, allocator: std.mem.Allocator ```
    pub fn Name(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KNSCore__CategoryMetadata_Name(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("knscore::categorymetadata.Name: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/knscore-categorymetadata.html#displayName)
    ///
    /// ``` self: QtC.KNSCore__CategoryMetadata, allocator: std.mem.Allocator ```
    pub fn DisplayName(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KNSCore__CategoryMetadata_DisplayName(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("knscore::categorymetadata.DisplayName: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KNSCore__CategoryMetadata ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KNSCore__CategoryMetadata_Delete(@ptrCast(self));
    }
};
